Pong codereview

FredTheDino opened the pull request.

FredTheDino committed:
Started commenting on the entity features

FredTheDino committed:
Don't copy res folder as it isn't needed anymore

FredTheDino committed:
Fix input inconsistency

sornas committed:
Add pong

FredTheDino committed:
Better error message when managing shapes.

FredTheDino committed:
Remove all traling whitespaces

FredTheDino committed:
Change mixed indentation

sornas committed:
Add paddles instead of ball

sornas committed:
Revert "Fix input inconsistency"

This reverts commit 91690c2e08939ae8a9ed51e6d3ae8bb6f025f166.

sornas committed:
Move Paddle.body location and init

FredTheDino committed:
Fix bug, shape id's wheren't being filled out.

sornas committed:
Crashing again, this time when checking collisions

sornas committed:
Actually include faulty code

sornas committed:
Remove calls to Function

sornas committed:
Merge remote-tracking branch 'origin/mac'

FredTheDino commented on src/game/game_main.cpp:
```diff
@@ -5,34 +5,71 @@
 
 namespace Game {
 
+const bool debug = true;
```
Maybe rename to "DEBUG"? To signify it's a constant.

sornas committed:
Fix vector<Paddle *>

FredTheDino commented on src/game/game_main.cpp:
```diff
@@ -48,24 +85,65 @@ void show_int(char *buffer, void *info) {
-    REGISTER_ENTITY(MyEnt);
 }
 
+std::vector<Paddle *> paddles = {};
```
You don't want a list of pointers. This causes you to access uninitialized memory. More comments further down.

FredTheDino commented on src/game/game_main.cpp:
```diff
@@ -48,24 +85,65 @@ void show_int(char *buffer, void *info) {
+    paddle.controllable = true;
+    paddle.player = Player::P1;
+    Logic::add_entity(paddle);
+    paddles.push_back(&paddle);
```
Here you push the address of the paddle, but it lives on the stack so it will be invalid after this function. This is causing the crash.

FredTheDino commented on src/game/game_main.cpp:
```diff
@@ -48,24 +85,65 @@ void show_int(char *buffer, void *info) {
+    paddle2.controllable = true;
+    paddle2.player = Player::P2;
+    Logic::add_entity(paddle2);
+    paddles.push_back(&paddle2);
```
Here you push the address of the paddle, but it lives on the stack so it will be invalid after this function. This is causing the crash.

Same as above.

FredTheDino commented on src/game/game_main.cpp:
```diff
@@ -76,7 +154,7 @@ void update(f32 delta) {
     Util::end_tweak_section(&show_camera_controls);
 
-    static bool show_entity_system = true;
+    static bool show_entity_system = false;
```
You can remove this tweak section, it was used to debug the entity system and then just left in here. My bad honestly.

FredTheDino commented on src/game/game_main.cpp:
```diff
@@ -85,23 +163,21 @@ void update(f32 delta) {
-        auto id = Logic::add_entity(e);
-        LOG("%d %d", id.slot, id.gen);
+    if (pressed(Name::LEFT)) {
+        Renderer::global_camera.position += V2(-1, 0);
```
This should probably contain some kind of delta time.

sornas replied:
Maybe not when checking pressed() instead of down(), but it was mainly used to test adding Vec2s together and then forgotten about.

FredTheDino commented on src/game/game_main.cpp:
```diff
@@ -85,23 +163,21 @@ void update(f32 delta) {
-        auto id = Logic::add_entity(e);
-        LOG("%d %d", id.slot, id.gen);
+    if (pressed(Name::RIGHT)) {
+        Renderer::global_camera.position += V2(1, 0);
```
This should probably contain some kind of delta time.

FredTheDino commented on src/game/game_main.cpp:
```diff
@@ -85,23 +163,21 @@ void update(f32 delta) {
     }
 
+    // Collisions
+    for (Paddle *paddle: paddles) {
```
Illegal memory access.

FredTheDino commented on src/game/game_main.cpp:
```diff
@@ -5,34 +5,71 @@
 
-Physics::ShapeID square;
+struct Paddle : public Logic::Entity {
+    Player player = Player::ANY;
```
I would group these with "REGISTER_FIELDS" so you don't have to scroll to access it.

sornas replied:
What happens when I register fields? Documentation only specifies that the fields are "exported" (https://fog.sörnäs.se/#entitysystemaddinganewentitytype).

FredTheDino commented on src/game/game_main.cpp:
```diff
@@ -48,24 +85,65 @@ void show_int(char *buffer, void *info) {
+    add(K(i), Name::UP,    Player::P2);
+    add(K(k), Name::DOWN,  Player::P2);
+
+    Vec2 ball_points[4] = {
```
You can remove the 4 here.

FredTheDino commented on src/game/game_main.cpp:
```diff
@@ -48,24 +85,65 @@ void show_int(char *buffer, void *info) {
+    };
+    ball_shape = Physics::add_shape(LEN(ball_points), ball_points);
 
+    Vec2 paddle_points[4] = {
```
You can remove the 4 here.

FredTheDino reviewed (commented):
I found some things that might be worth changing. Let me know if anything is unclear.

sornas committed:
Code review

sornas committed:
Add comparision between body global and in struct

sornas committed:
Add id to structs (not working)

Entities disappear when Logic::EntityID is added to the respective
struct, even if it's never initialized.

sornas committed:
Working collisions

sornas committed:
Stop on collision and draw only debug-physics

sornas committed:
Actual gameplay!

sornas committed:
Points!

sornas committed:
Basic particles

Cool and good

sornas committed:
Play sound on hit

sornas committed:
Bounce against (invisible) walls

Basically I'm just testing as many parts of the engine as possible.
